src/biguint.rs

```diff
@@ -2255,6 +2266,200 @@ pub(crate) fn to_str_radix_reversed(u: &BigUint, radix: u32) -> Vec<u8> {
     res
 }
 
+#[cfg(u64_digit)]
+pub struct IterU32Digits<'a> {
+    data: &'a [u64],
+    next_is_lo: bool,
+    last_hi_is_zero: bool,
+}
+#[cfg(u64_digit)]
+impl<'a> IterU32Digits<'a> {
+    fn new(data: &'a [u64]) -> Self {
+        let last_hi_is_zero = data
+            .last()
+            .map(|&last| {
+                let last_hi = (last >> 32) as u32;
+                last_hi == 0
+            })
+            .unwrap_or(false);
+        IterU32Digits {
+            data,
+            next_is_lo: true,
+            last_hi_is_zero,
+        }
+    }
+}
+#[cfg(u64_digit)]
+impl<'a> Iterator for IterU32Digits<'a> {
```

Comment:
Similarly on impls, the lifetime can be omitted when it's not otherwise named.
```suggestion
impl Iterator for IterU32Digits<'_> {
```

Reply:
This too, there are several other impls like it that can omit the lifetime.